src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -5954,10 +5993,12 @@ private static void ReportDuplicateObjectMemberInitializers(BoundExpression boun
 
                 var memberInitializerSyntax = boundMemberInitializer.Syntax;
 
-                Debug.Assert(memberInitializerSyntax.Kind() == SyntaxKind.SimpleAssignmentExpression);
-                var namedAssignment = (AssignmentExpressionSyntax)memberInitializerSyntax;
+                var memberNameSyntax = memberInitializerSyntax as IdentifierNameSyntax;
+                if (memberInitializerSyntax is AssignmentExpressionSyntax assignment)
+                {
+                    memberNameSyntax = assignment.Left as IdentifierNameSyntax;
+                }
```

Comment:
Feels like we should still be maintaining the original invariants here, just adjusted. For example:
```suggestion
                IdentifierNameSyntax memberNameSyntax;
                if (memberInitializerSyntax is AssignmentExpressionSyntax namedAssignment)
                {
                    Debug.Assert(memberInitializerSyntax.Kind() == SyntaxKind.SimpleAssignmentExpression);
                    memberNameSyntax = namedAssignment.Left as IdentifierNameSyntax;
                }
                else
                {
                    memberNameSyntax = (IdentifierNameSyntax)memberInitializerSyntax;
                }
```